### PR TITLE
add information on Assistant remediation guidance

### PR DESCRIPTION
**Pull request by khorne3**
I opted to replace the current Assistant autofix section with a remediation section that's further broken down into guidance and autofix.

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team

### CODE REVIEW

File: docs/semgrep-assistant/overview.md [hunk 1]
```diff
@@ -65,26 +65,39 @@ Assistant's suggestions to ignore findings are also surfaced in PR or MR comment
...
+Semgrep Assistant can provide remediation advice and autofixes, or suggested fixes, for Semgrep Code findings.
 
-You can set the minimum autofix confidence level required to display autofix suggestions from Semgrep Assistant in Semgrep AppSec Platform's **Settings** page. To receive as many Assistant suggestions as are available, set the minimum to **low confidence**.
+#### Guidance
 
-Autofixes are available in PR and MR comments, so developers can review and verify Semgrep's generated fixes before they're applied.
+With Assistant enabled, every PR or MR comment pushed by Semgrep contains remediation guidance unless the original rule message contains information on how to fix the issue. Assistant's remediation guidance provides step-by-step instructions on how to remediate the finding identified by Semgrep Code.
```
Comment (s-santillan) [inline docs/semgrep-assistant/overview.md:74]: > "Unless the original rule message contains information..." 

Does this mean, if there's any content in `message` or if the `message` key is used, there is no remediation guidance?

File: docs/semgrep-assistant/overview.md [hunk 2]
```diff
@@ -65,26 +65,39 @@ Assistant's suggestions to ignore findings are also surfaced in PR or MR comment
...
+
+![PR comments with remediation advice](/img/assistant-guidance.png)
+***Figure***. PR comment displaying the rule message followed by a comment that contains Assistant-generated remediation guidance.
+
+#### Autofix
+
+Semgrep Assistant can suggest [autofix](/writing-rules/autofix/) code snippets for Semgrep Code findings when it identifies a true positive. Assistant only suggests an autofix if the rule doesn't have a human-written autofix.
```
Comment (s-santillan) [inline docs/semgrep-assistant/overview.md:81]:
```suggestion
Semgrep Assistant can suggest [autofix](/writing-rules/autofix/) code snippets for Semgrep Code findings when it identifies a true positive. Assistant only suggests an autofix if the rule doesn't have a human-written autofix. 
```

So the conditions for AI autofix are:
- Setting is enabled
- No content or value in `fix` (human written autofix)
- Confidence level set by user is met

Yes?

File: docs/semgrep-assistant/overview.md (hunk 1 above)
Comment (s-santillan) [reply docs/semgrep-assistant/overview.md:74]: Basically, is it `message`, `fix`, or **both **?

**Review by s-santillan — APPROVED** [review]

File: docs/semgrep-assistant/overview.md (hunk 2 above)
Comment (jaksiks) [reply docs/semgrep-assistant/overview.md:81]: Correct!

File: docs/semgrep-assistant/overview.md (hunk 1 above)
Comment (jaksiks) [reply docs/semgrep-assistant/overview.md:74]: No, so we will attempt to generate remediation guidance for all issues. We then do a semantic similarity check to make sure we aren't posting a comment that just echos what was said in the rule message.

File: docs/semgrep-assistant/overview.md [hunk 3]
```diff
@@ -65,26 +65,39 @@ Assistant's suggestions to ignore findings are also surfaced in PR or MR comment
...
+
+You can set the minimum autofix confidence level required to display autofix suggestions from Semgrep Assistant on Semgrep AppSec Platform's **Settings** page. To receive as many Assistant suggestions as are available, set the minimum to **low confidence**.
+
+Autofixes are available in PR and MR comments, so developers can review and verify Semgrep's generated fixes before applying them.
 
 ![Semgrep Assistant generating a potential fix in a comment](/img/semgrep-assistant-autofix.png)
 *Figure*. Semgrep Assistant generates a potential fix in a PR comment.
 
-Autofixes are also available in Semgrep AppSec Platform's **Findings page** under **Your code** in the [finding's details](/docs/semgrep-code/findings/#view-findings-details-about-a-specific-finding). 
+Autofixes are also available on Semgrep AppSec Platform's **Findings page** under **Your code** in the [finding's details](/docs/semgrep-code/findings/#view-findings-details-about-a-specific-finding).
```
Comment (jaksiks) [inline docs/semgrep-assistant/overview.md:92]: Remediation is also available on the finding details page

File: static/img/assistant-guidance.png
Comment (jaksiks) [inline static/img/assistant-guidance.png:1]: Let me find a more recent example with our updated formatting.

File: docs/semgrep-assistant/overview.md (hunk 1 above)
Comment (khorne3) [reply docs/semgrep-assistant/overview.md:74]: I'll simplify to saying that we'll generate guidance for all issues; I don't think we need to provide further details on how we choose wording unless you think it necessary?

File: docs/semgrep-assistant/overview.md (hunk 2 above)
Comment (khorne3) [reply docs/semgrep-assistant/overview.md:81]: Sara, I reorganized the paragraphs so those bits of information aren't as spread out

File: static/img/assistant-guidance.png
Comment (khorne3) [reply static/img/assistant-guidance.png:1]: I found one (and some of the UI) and swapped out the screenshots.

**Review by jaksiks — APPROVED** [review]

**Pull request merged**